subscribers.js

```diff
@@ -1,7 +1,89 @@
 "use strict";
 
-// Set of all registered emails.
-// TODO: Implement a real persistent storage solution.
-const gEmails = new Set();
+const pg = require("pg");
 
-module.exports = gEmails;
+const dbConfig = {
+  max: 10,
+};
+
+const dbPool = new pg.Pool(dbConfig);
+
+function dbq(aQuery) {
+  return dbPool.query(aQuery);
+}
+
+function _ret(aError=null, aOther={}) {
+  return Object.assign({
+    error: aError,
+  }, aOther);
+}
+
+const Subscribers = {
+  async addUser(aEmail) {
+    // Error code for attempting to add a duplicate entry on a UNIQUE key.
+    const DUPLICATE_ERROR = "23505";
+    try {
+      await dbq({
+        text: "INSERT INTO users ( email ) VALUES ( $1 );",
+        values: [ aEmail ],
+      });
+
+      return _ret();
+    } catch (e) {
+      if (e.code === DUPLICATE_ERROR) {
+        // Duplicate entry, count it as success.
+        return _ret(null, { duplicate: true });
+      }
+
+      return _ret(e);
+    }
+  },
+
+  async deleteUser(aEmail) {
+    try {
+      await dbq({
+        text: "DELETE FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      });
+      return _ret();
+    } catch(e) {
+      return _ret(e);
+    }
+  },
+
+  async getUser(aEmail) {
+    try {
+      const rows = (await dbq({
+        text: "SELECT * FROM users WHERE email = $1;",
+        values: [ aEmail ],
+      })).rows;
+
+      if (!rows.length) {
+        return _ret(new Error("Email not found"));
+      }
+
+      const { email } = rows[0];
+      if (!email) {
+        // This shouldn't happen, but... better safe than sorry.
+        return _ret(new Error("Entry was found but with no email value"));
+      }
+
+      return _ret(null, { email });
+    } catch(e) {
+      return _ret(e);
+    }
+  },
+
+  async clearAllUsers() {
```

Comment:
Does anybody call this API endpoint?

I see we removed `clearUserList()` and `dumpUserList()`, but am not convinced this is even remotely related to those public/js/test.js methods:

```diff
-// eslint-disable-next-line no-unused-vars	
-function clearUserList() {	
-  doXHR("/user/reset", {});	
-}	
-	
-// eslint-disable-next-line no-unused-vars	
-function dumpUserList() {	
-  doXHR("/user/list", {});	
-}
```